src/lib/profiles/data-management/Current/MessageDef.cpp

```diff
@@ -2741,13 +2746,19 @@ WEAVE_ERROR StatusList::Parser::CheckSchemaValidity(void) const
 
     while (WEAVE_NO_ERROR == (err = reader.Next()))
     {
+        TLVType type;
+        uint64_t tag;
+
         // TODO: The spec says the StatusList should be an array of arrays, but in
         // the current implementation it's an array of structures. The array of
         // arrays is less intuitive but more space efficient.
-        VerifyOrExit(nl::Weave::TLV::AnonymousTag == reader.GetTag(), err = WEAVE_ERROR_INVALID_TLV_TAG);
 
-        VerifyOrExit((nl::Weave::TLV::kTLVType_Structure == reader.GetType() ||
-                      nl::Weave::TLV::kTLVType_Array), err = WEAVE_ERROR_WRONG_TLV_TYPE);
+        tag = reader.GetTag();
+        VerifyOrExit(nl::Weave::TLV::AnonymousTag == tag, err = WEAVE_ERROR_INVALID_TLV_TAG);
```

Comment:
Note that the right side of the OR (||) in 👍 

```
(nl::Weave::TLV::kTLVType_Structure == reader.GetType() || nl::Weave::TLV::kTLVType_Array)
```

is effectively always true. The solve is either:

```
(nl::Weave::TLV::kTLVType_Structure == reader.GetType() || nl::Weave::TLV::kTLVType_Array == reader.GetType())
```

and hope there are no side effects to reader.GetType(). But that seems sloppy anyway, so that leaves reading type to a local and then verifying:

```
type = reader.GetType();
(nl::Weave::TLV::kTLVType_Structure == type || nl::Weave::TLV::kTLVType_Array == type)
```

Frankly, I dislike the above coding style anyway and would have done:

```
type = reader.GetType();
(type == nl::Weave::TLV::kTLVType_Structure || type == nl::Weave::TLV::kTLVType_Array)
```

but I tried to keep most of these fixes surgical.